### PR TITLE
separate parameter lifetime from token credential lifetime

### PR DESCRIPTION
**Pull request by esheppa**
In the current API, the lifetime of the `T: TokenCredential` stored within the `KeyClient` is `'a` and all parameters to functions that take a `secret_name` or similar, are also using that `'a`. This can make the current API tricky to use, however there is no issue to separate these lifetimes, although there are other ways to solve this that may be preferable, for example storing an owned `TokenCredential` in the `KeyClient`.  The advantage of this change is that it doesn't break API compatibility, but storing an owned `TokenCredential` may be more ergonomic (this may require a discussion on whether `TokenCredential` should require `Clone`).

### CODE REVIEW

File: sdk/key_vault/src/secret.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ impl<'a, T: TokenCredential> KeyClient<'a, T> {
     ///
     /// Runtime::new().unwrap().block_on(example());
     /// ```
-    pub async fn get_secret(&mut self, secret_name: &'a str) -> Result<KeyVaultSecret, Error> {
+    pub async fn get_secret<'b>(&mut self, secret_name: &'b str) -> Result<KeyVaultSecret, Error> {
```
Comment (yoshuawuyts) [inline sdk/key_vault/src/secret.rs:141]: This does not require manual lifetime ascription, instead we should do:
```suggestion
    pub async fn get_secret(&mut self, secret_name: &str) -> Result<KeyVaultSecret, Error> {
```

**Review by yoshuawuyts — CHANGES_REQUESTED** [review]
Hey, thanks for opening this! You're indeed right that binding the lifetimes of arguments to the lifetime of the struct makes no sense. However because of that, it appears we don't need to annotate any lifetimes at all!

I've included an example of how we could simplify this further in a code snippet. If you could try changing the remaining annotations in a similar way, that'd be great.

Thanks so much!

File: sdk/key_vault/src/secret.rs (hunk 1 above)
Comment (esheppa) [reply sdk/key_vault/src/secret.rs]: looks good, I've applied this in all locations

**Comment by esheppa** [comment]
also added an example to show a use of `KeyClient` that wasn't possible prior to this change

**Review by yoshuawuyts — APPROVED** [review]
This looks great. Thanks!

**Comment by yoshuawuyts** [comment]
@esheppa CI is failing because of a missing `cargo fmt`. If you could run that we should be good to merge!

**Review by yoshuawuyts — APPROVED** [review]
Thanks heaps!

**Pull request merged**